## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 442446`
Commit: [`4b1bdd1a4babfa1d522aa3bd1832ef3a4f1aba4b`](https://github.com/NixOS/nixpkgs/commit/4b1bdd1a4babfa1d522aa3bd1832ef3a4f1aba4b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/4b1bdd1a4babfa1d522aa3bd1832ef3a4f1aba4b..pull/442446/head))
Merge: [`67775b9a4eea283b74c716a258401aceead8eeff`](https://github.com/NixOS/nixpkgs/commit/67775b9a4eea283b74c716a258401aceead8eeff)

Logs: https://github.com/lonerOrz/nixpkgs-review-gha/actions/runs/17691287566


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qwen-code</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qwen-code</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qwen-code</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>qwen-code</summary>
<pre>npm error       | ^~~~~~~~~~~~~~~~~~~~~~
npm error /nix/store/ha64ikws19bqb6s08qcmv6ybqjza3gki-nodejs-22.19.0-source/src/node.h:1262:3: note: expanded from macro &#x27;NODE_MODULE&#x27;
npm error  1262 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
npm error       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error /nix/store/ha64ikws19bqb6s08qcmv6ybqjza3gki-nodejs-22.19.0-source/src/node.h:1228:7: note: expanded from macro &#x27;NODE_MODULE_X&#x27;
npm error  1228 |       (node::addon_register_func) (regfunc),                          \
npm error       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error 3 warnings and 1 error generated.
npm error make: *** [pty.target.mk:124: Release/obj.target/pty/src/unix/pty.o] Error 1
npm error gyp ERR! build error 
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.&lt;anonymous&gt; (/nix/store/xgqkmyswq6s9jbdhvx6x8ix2jkj0jbip-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:219:23)
npm error gyp ERR! System Darwin 24.6.0
npm error gyp ERR! command &quot;/nix/store/xgqkmyswq6s9jbdhvx6x8ix2jkj0jbip-nodejs-22.19.0/bin/node&quot; &quot;/nix/store/xgqkmyswq6s9jbdhvx6x8ix2jkj0jbip-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js&quot; &quot;rebuild&quot;
npm error gyp ERR! cwd /nix/build/nix-build-qwen-code-0.0.11.drv-7977-3415684879/source/node_modules/node-pty
npm error gyp ERR! node -v v22.19.0
npm error gyp ERR! node-gyp -v v11.2.0
npm error gyp ERR! not ok
npm error Log files were not written due to an error writing to the directory: /nix/store/d4sk7bvldgw2j92hg8rs4xdm8zvf3n0m-qwen-code-0.0.11-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qwen-code</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>qwen-code</summary>
<pre>npm error       | ^~~~~~~~~~~~~~~~~~~~~~
npm error /nix/store/rmms24dkf1z5fdj2lfx9ln79r8lkm2aq-nodejs-22.19.0-source/src/node.h:1262:3: note: expanded from macro &#x27;NODE_MODULE&#x27;
npm error  1262 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
npm error       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error /nix/store/rmms24dkf1z5fdj2lfx9ln79r8lkm2aq-nodejs-22.19.0-source/src/node.h:1228:7: note: expanded from macro &#x27;NODE_MODULE_X&#x27;
npm error  1228 |       (node::addon_register_func) (regfunc),                          \
npm error       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm error 3 warnings and 1 error generated.
npm error make: *** [pty.target.mk:124: Release/obj.target/pty/src/unix/pty.o] Error 1
npm error gyp ERR! build error 
npm error gyp ERR! stack Error: `make` failed with exit code: 2
npm error gyp ERR! stack at ChildProcess.&lt;anonymous&gt; (/nix/store/3scj3c1qxh5iclf8jy5hbl9pki40y77n-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:219:23)
npm error gyp ERR! System Darwin 24.6.0
npm error gyp ERR! command &quot;/nix/store/3scj3c1qxh5iclf8jy5hbl9pki40y77n-nodejs-22.19.0/bin/node&quot; &quot;/nix/store/3scj3c1qxh5iclf8jy5hbl9pki40y77n-nodejs-22.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js&quot; &quot;rebuild&quot;
npm error gyp ERR! cwd /nix/build/nix-build-qwen-code-0.0.11.drv-7668-3460493970/source/node_modules/node-pty
npm error gyp ERR! node -v v22.19.0
npm error gyp ERR! node-gyp -v v11.2.0
npm error gyp ERR! not ok
npm error Log files were not written due to an error writing to the directory: /nix/store/d4sk7bvldgw2j92hg8rs4xdm8zvf3n0m-qwen-code-0.0.11-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal</pre>
</details>
</details>

--- a/pkgs/by-name/qw/qwen-code/add-missing-resolved-integrity-fields.patch
+++ b/pkgs/by-name/qw/qwen-code/add-missing-resolved-integrity-fields.patch
@@ -1,0 +1,70 @@
+From 0735b59eba183574e080d2912347d58e6e3d158e Mon Sep 17 00:00:00 2001
+From: Sergey Volkov <taranarmo@gmail.com>
+Date: Fri, 12 Sep 2025 20:06:24 +0200
+Subject: [PATCH] Add missing resolved and integrity fields to several packages
+
+---
+ package-lock.json | 12 ++++++++++++
+ 1 file changed, 12 insertions(+)
+
+diff --git a/package-lock.json b/package-lock.json
+index 146686dc..9ad9009f 100644
+--- a/package-lock.json
++++ b/package-lock.json
+@@ -12574,6 +12574,8 @@
+     },
+     "packages/cli/node_modules/@testing-library/dom": {
+       "version": "10.4.0",
++      "resolved": "https://registry.npmjs.org/@testing-library/dom/-/dom-10.4.0.tgz",
++      "integrity": "sha512-pemlzrSESWbdAloYml3bAJMEfNh1Z7EduzqPKprCH5S341frlpYnUEW0H72dLxa6IsYr+mPno20GiSm+h9dEdQ==",
+       "dev": true,
+       "license": "MIT",
+       "peer": true,
+@@ -12609,6 +12611,8 @@
+     },
+     "packages/cli/node_modules/@testing-library/react": {
+       "version": "16.3.0",
++      "resolved": "https://registry.npmjs.org/@testing-library/react/-/react-16.3.0.tgz",
++      "integrity": "sha512-kFSyxiEDwv1WLl2fgsq6pPBbw5aWKrsY2/noi1Id0TK0UParSF62oFQFGHXIyaG4pp2tEub/Zlel+fjjZILDsw==",
+       "dev": true,
+       "license": "MIT",
+       "dependencies": {
+@@ -12660,6 +12664,8 @@
+     },
+     "packages/cli/node_modules/aria-query": {
+       "version": "5.3.0",
++      "resolved": "https://registry.npmjs.org/aria-query/-/aria-query-5.3.0.tgz",
++      "integrity": "sha512-b0P0sZPKtyu8HkeRAfCq0IfURZK+SuwMjY1UXGBU27wpAiTwQAIlq56IbIO+ytk/JjS1fMR14ee5WBBfKi5J6A==",
+       "dev": true,
+       "license": "Apache-2.0",
+       "peer": true,
+@@ -12669,6 +12675,8 @@
+     },
+     "packages/cli/node_modules/emoji-regex": {
+       "version": "10.4.0",
++      "resolved": "https://registry.npmjs.org/emoji-regex/-/emoji-regex-10.4.0.tgz",
++      "integrity": "sha512-EC+0oUMY1Rqm4O6LLrgjtYDvcVYTy7chDnM4Q7030tP4Kwj3u/pR6gP9ygnp2CJMK5Gq+9Q2oqmrFJAz01DXjw==",
+       "license": "MIT"
+     },
+     "packages/cli/node_modules/react-is": {
+@@ -12681,6 +12689,8 @@
+     },
+     "packages/cli/node_modules/string-width": {
+       "version": "7.2.0",
++      "resolved": "https://registry.npmjs.org/string-width/-/string-width-7.2.0.tgz",
++      "integrity": "sha512-tsaTIkKW9b4N+AEj+SVA+WhJzV7/zMhcSu78mLKWSk7cXMOSHsBKFWUs0fWwq8QyK3MgJBQRX6Gbi4kYbdvGkQ==",
+       "license": "MIT",
+       "dependencies": {
+         "emoji-regex": "^10.3.0",
+@@ -12815,6 +12825,8 @@
+     },
+     "packages/core/node_modules/ignore": {
+       "version": "7.0.5",
++      "resolved": "https://registry.npmjs.org/ignore/-/ignore-7.0.5.tgz",
++      "integrity": "sha512-Hs59xBNfUIunMFgWAbGX5cq6893IbWg4KnrjbYwX3tx0ztorVgTDA6B2sxf8ejHJ4wz8BqGUMYlnzNBer5NvGg==",
+       "license": "MIT",
+       "engines": {
+         "node": ">= 4"
+-- 
+2.51.0
+

--- a/pkgs/by-name/qw/qwen-code/package.nix
+++ b/pkgs/by-name/qw/qwen-code/package.nix
@@ -2,25 +2,56 @@
   lib,
   buildNpmPackage,
   fetchFromGitHub,
-  fetchNpmDeps,
   nix-update-script,
+  jq,
 }:
 
 buildNpmPackage (finalAttrs: {
   pname = "qwen-code";
-  version = "0.0.6";
+  version = "0.0.11";
 
   src = fetchFromGitHub {
     owner = "QwenLM";
     repo = "qwen-code";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-s4+1hqdlJh5jOy6uZz608n5DzuBR+v/s+7D85oFwQIY=";
+    hash = "sha256-5qKSWbc0NPpgvt36T/gRSgm1+o2Pbdw3tgfcGba6YSs=";
   };
 
-  npmDeps = fetchNpmDeps {
-    inherit (finalAttrs) src;
-    hash = "sha256-cGO66hQxgpoxphtt/BPPDIBuAG8yQseCdzUdAO2mkr4=";
-  };
+  patches = [
+    # similar to upstream gemini-cli some node deps are missing resolved and integrity fields
+    # upstream the problem is solved in master and in v0.4+, eventually the fix should arrive to qwen
+    ./add-missing-resolved-integrity-fields.patch
+  ];
+
+  npmDepsHash = "sha256-XvJO3ylm/ER5neSubci2w9OCTmqobmmXLbKmdQAqArY=";
+
+  nativeBuildInputs = [
+    jq
+  ];
+
+  postPatch = ''
+    # patches below remove node-pty dependency which causes build fail on Darwin
+    # should be conditional on platform but since package-lock.json is patched it changes its hash
+    # though seems like this dependency is not really required by the package
+    ${jq}/bin/jq '
+      del(.packages."node_modules/node-pty") |
+      del(.packages."node_modules/@lydell/node-pty") |
+      del(.packages."node_modules/@lydell/node-pty-darwin-arm64") |
+      del(.packages."node_modules/@lydell/node-pty-darwin-x64") |
+      del(.packages."node_modules/@lydell/node-pty-linux-arm64") |
+      del(.packages."node_modules/@lydell/node-pty-linux-x64") |
+      del(.packages."node_modules/@lydell/node-pty-win32-arm64") |
+      del(.packages."node_modules/@lydell/node-pty-win32-x64") |
+      walk(
+        if type == "object" and has("dependencies") then
+          .dependencies |= with_entries(select(.key | contains("node-pty") | not))
+        elif type == "object" and has("optionalDependencies") then
+          .optionalDependencies |= with_entries(select(.key | contains("node-pty") | not))
+        else .
+        end
+      )
+    ' package-lock.json > package-lock.json.tmp && mv package-lock.json.tmp package-lock.json
+  '';
 
   buildPhase = ''
     runHook preBuild
@@ -34,10 +65,10 @@ buildNpmPackage (finalAttrs: {
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin
-    cp -r bundle/* $out/
-    patchShebangs $out
-    ln -s $out/gemini.js $out/bin/qwen
+    mkdir -p $out/bin $out/share/qwen-code
+    cp -r bundle/* $out/share/qwen-code/
+    patchShebangs $out/share/qwen-code
+    ln -s $out/share/qwen-code/gemini.js $out/bin/qwen
 
     runHook postInstall
   '';

